src/main/java/oncall/controller/OncallController.java

```diff
@@ -0,0 +1,79 @@
+package oncall.controller;
+
+import static oncall.messages.ErrorMessages.INVALID_INPUT;
+
+import oncall.domain.Employees;
+import oncall.domain.date.Dates;
+import oncall.domain.date.Oncall;
+import oncall.domain.date.StartDate;
+import oncall.domain.dto.OncallDto;
+import oncall.mapper.OncallMapper;
+import oncall.service.OncallService;
+import oncall.util.InputUtil;
+import oncall.util.Pair;
+import oncall.view.InputView;
+import oncall.view.OutputView;
+
+public class OncallController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final OncallService oncallService;
+
+    public OncallController(final InputView inputView,
+                            final OutputView outputView,
+                            final OncallService oncallService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.oncallService = oncallService;
+    }
+
+    public void play() {
+        final StartDate startDate = inputValidStartDate();
+        final Dates dates = oncallService.generateDates(startDate);
+
+        final Pair<Employees, Employees> employeesPair = inputValidWorkdayAndHolidayEmployees();
+        final Employees workdayEmployees = employeesPair.getFirst();
+        final Employees holidayEmployees = employeesPair.getSecond();
+
+        final Oncall oncall = oncallService.generateOncall(
+                startDate, dates,
+                workdayEmployees,
+                holidayEmployees);
+
+        outputResult(startDate, dates, oncall);
+    }
+
+    private StartDate inputValidStartDate() {
+        return InputUtil.retryOnInvalidInput(inputView::inputStartDate,
+                errorMessage -> outputView.outputErrorMessage(INVALID_INPUT.getMessage()));
+    }
+
+    private Employees inputValidWorkdayEmployees() {
+        return InputUtil.retryOnInvalidInput(inputView::inputWorkdayEmployees,
+                errorMessage -> outputView.outputErrorMessage(INVALID_INPUT.getMessage()));
+    }
+
+    private Employees inputValidHolidayEmployees() {
+        return InputUtil.retryOnInvalidInput(inputView::inputHolidayEmployees,
+                errorMessage -> outputView.outputErrorMessage(INVALID_INPUT.getMessage()));
+    }
+
+    private Pair<Employees, Employees> inputValidWorkdayAndHolidayEmployees() {
+        while (true) {
+            try {
+                final Employees workdayEmployees = inputView.inputWorkdayEmployees();
+                final Employees holidayEmployees = inputView.inputHolidayEmployees();
+                return new Pair<>(workdayEmployees, holidayEmployees);
+            } catch (IllegalArgumentException e) {
+                outputView.outputErrorMessage(INVALID_INPUT.getMessage());
+            }
+        }
+    }
```

Comment:
이 부분은 저 역시 제 코드에 아쉬움을 느끼는 부분이라 코멘트 남깁니다.
평일 근무 순번과 주말 근무 순번이 현재는 순서로 구분되어 있는 상황입니다. 즉, 평일 근무 순번은 `getFirst`로 가져와야 합니다. 이 때문에 코드 전체에서 "평일 근무는 첫번째, 주말 근무는 두번째" 라는 정보를 알고 있어야 하는데요.
이런 불편함을 해소하기 위해선 근무 순번들을 저장하고 있는 클래스 `TotalEmployees`가 `getWeekdayEmployees()` 메서드를 제공해 주면 훨씬 편해질 것 같아요.

하지만, 여기까지만 생각하면 결국 `TotalEmployees` 클래스의 생성시점에 생성자에 전달해 줄 파라미터의 순서를 알고 있어야 하기에 똑같은 불편함이 여전히 남아있습니다. 
이를 해결하기 위해 빌더 패턴을 함께 사용한다면 훨씬 코드 사용이 편리해 질 것 같아요.

아래는 `TotalEmployees`와 빌더패턴이 구현되었다는 가정 하에 작성될 코드 예시입니다.
```suggestion
    private TotalEmployees inputValidWorkdayAndHolidayEmployees() {
        while (true) {
            try {
                final Employees workdayEmployees = inputView.inputWorkdayEmployees();
                final Employees holidayEmployees = inputView.inputHolidayEmployees();
                return TotalEmployees.builder()
                    .withWorkdayEmployees(workdayEmployees)
                    .withHolidayEmployees(holidayEmployees)
                    .build();
            } catch (IllegalArgumentException e) {
                outputView.outputErrorMessage(INVALID_INPUT.getMessage());
            }
        }
    }
```

Reply:
빌더 패턴 너무 좋은 방법이네요 👍 
제 개인적인 취향으로는 빌더 패턴은 매개변수가 4개 이상일 때 쓰는 것이 좋다고 생각합니다! ㅎㅎ 그 이하는 빌더패턴을 관리하는 비용이 더 크다고 생각하기 때문이에요!
다만 추후 확장성을 고려하면 여기서 Pair로 두는 것보다는 TotalEmployees 로 감싸는 것은 확실히 좋은 방법이라 생각해요! 👍

또, 장수님이 쓰신 코드 중에 람다를 이용해서 함수형 인터페이스에 입력 함수 전달하는 것도 너무 좋아보이더라구요 ㅎㅎ
```java
    private TotalEmployees inputValidEmployees() {
        return InputUtil.retryOnInvalidInput(() -> inputEmployees(),
                errorMessage -> outputView.outputErrorMessage(INVALID_INPUT.getMessage()));
    }

    private TotalEmployees inputEmployees() {
        final Employees workdayEmployees = inputView.inputWorkdayEmployees();
        final Employees holidayEmployees = inputView.inputHolidayEmployees();
        return TotalEmployees.create(workdayEmployees,holidayEmployees)
    }
```
그래서 이렇게 리팩터링 해봤습니다 ㅎㅎ 좋은 리뷰 감사합니다!👍

Reply:
Pair 쓰는 코드 보고 `왜 이렇게 하셨지...` 한참 고민했습니다 ㅋㅋ

> 평일 순번 또는 휴일 순번의 입력 값이 올바르지 않은 경우, '평일 순번'부터 다시 입력 받는다.

이 제약 조건 때문이었나보군요.
적절하게 리팩터링 하신 것 같네요 👍👍👍